src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -136,6 +138,26 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
         }
   }
 
+  override def simulateTransaction(stx: SignedTransaction, blockHeader: BlockHeader, storages: BlockchainStorages, validators: Validators): TxResult = {
+    val blockchain = BlockchainImpl(storages)
+    val stateRoot = blockchain.getBlockHeaderByHash(blockHeader.hash).map(_.stateRoot)
+    val world = InMemoryWorldStateProxy(storages, stateRoot)
+
+    val gasLimit = stx.tx.gasLimit
+    val config = EvmConfig.forBlock(blockHeader.number, blockchainConfig)
+
+    val context: PC = stx.tx.receivingAddress match {
```

Comment:
I'm wondering, could we use the `prepareProgramContext` function to create this `ProgramContext`? The only difference that I see is that we currently don't transfer the tx value to the receiver of the tx, should we transfer it?